app/src/main/java/org/mozilla/fenix/settings/WallpaperSettingsFragment.kt

```diff
@@ -0,0 +1,175 @@
+    ): View? {
+        binding.composeView.apply {
+            setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+            setContent {
+                FirefoxTheme {
+                    WallpaperSettings(wallpaperManager)
+                }
+            }
+        }
+        return binding.root
+    }
+}
+
+@Composable
+fun WallpaperSettings(
+    wallpaperManager: WallpaperManager
+) {
+    val currentlySelectedWallpaper = wallpaperManager.currentWallpaper.collectAsState()
+    Column {
+        Surface {
+            WallpaperThumbnails(
+                wallpapers = wallpaperManager.wallpapers.toList(),
+                onSelectionChanged = { wallpaperManager.updateWallpaperSelection(it) },
+                selectedWallpaper = currentlySelectedWallpaper.value
+            )
+        }
+    }
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun WallpaperThumbnails(
+    wallpapers: List<Wallpaper>,
+    onSelectionChanged: (Wallpaper) -> Unit,
+    selectedWallpaper: Wallpaper,
+    numColumns: Int = 3,
+) {
+    LazyVerticalGrid(
+        cells = GridCells.Fixed(numColumns),
+        modifier = Modifier.padding(vertical = 30.dp, horizontal = 20.dp)
+    ) {
+        items(wallpapers) { wallpaper ->
+            WallpaperThumbnailItem(
+                wallpaper = wallpaper,
+                onSelectionChanged = onSelectionChanged,
+                isCurrentlySelected = selectedWallpaper == wallpaper
+            )
+        }
+    }
+}
+
+@Composable
+fun WallpaperThumbnailItem(
+    wallpaper: Wallpaper,
+    onSelectionChanged: (Wallpaper) -> Unit,
+    isCurrentlySelected: Boolean,
+    aspectRatio: Float = 1.1f
+) {
+    val thumbnailShape = RoundedCornerShape(8.dp)
+    val border = if (isCurrentlySelected) {
+        Modifier.border(
+            BorderStroke(width = 2.dp, color = FirefoxTheme.colors.borderSelected),
+            thumbnailShape
+        )
+    } else {
+        Modifier.border(
+            BorderStroke(width = 1.dp, color = FirefoxTheme.colors.borderDivider),
+            thumbnailShape
+        )
+    }
+    val background = if (wallpaper == Wallpaper.NONE) {
+        Modifier.background(color = FirefoxTheme.colors.layer1)
+    } else {
+        Modifier
+    }
+    Box(
+        modifier = Modifier
+            .fillMaxWidth()
+            .aspectRatio(aspectRatio)
+            .padding(4.dp)
+            .clip(thumbnailShape)
+            .then(background)
+            .then(border)
+            .clickable { onSelectionChanged(wallpaper) }
+    ) {
+        if (wallpaper != Wallpaper.NONE) {
+            WallpaperImageThumbnail(
+                wallpaper = wallpaper,
+            )
+        }
+    }
+}
+
+@Composable
+fun WallpaperImageThumbnail(
+    wallpaper: Wallpaper,
+) {
+    val localizedContentDescription = LocalContext.current.resources.getString(
+        R.string.content_description_wallpaper_name, wallpaper.name
+    )
```

Comment:
This look strangely verbose to me since I recall an easier way to get the string resource. Perhaps try `stringResource`

Reply:
That's the one, thanks. Shortened the unnecessarily verbose name while I was at it